node/src/accountant/db_access_objects/receivable_dao.rs

```diff
@@ -724,92 +830,349 @@ mod tests {
+        let logger = Logger::new(test_name);
         let mut subject = ReceivableDaoReal::new(
             DbInitializerReal::default()
                 .initialize(&home_dir, DbInitializationConfig::test_default())
                 .unwrap(),
         );
-        let transactions = vec![BlockchainTransaction {
-            from: debtor.clone(),
-            wei_amount: 2300_u128,
-            block_number: 33_u64,
-        }];
+        subject
+            .more_money_receivable(
+                previous_timestamp,
+                &first_tracked_wallet,
+                first_initial_balance,
+            )
+            .unwrap();
+        subject
+            .more_money_receivable(
+                previous_timestamp,
+                &second_tracked_wallet,
+                second_initial_balance,
+            )
+            .unwrap();
+        subject.logger = logger;
+        let transaction_1 = BlockchainTransaction {
+            block_number: 4444,
+            from: first_tracked_wallet.clone(),
+            wei_amount: 1111,
+        };
+        let transaction_2 = BlockchainTransaction {
+            block_number: 4446,
+            from: unknown_wallet.clone(),
+            wei_amount: 2222,
+        };
+        let transaction_3 = BlockchainTransaction {
+            block_number: 4450,
+            from: second_tracked_wallet.clone(),
+            wei_amount: 9999,
+        };
+        let transactions = vec![transaction_1, transaction_2, transaction_3];
 
-        subject.more_money_received(SystemTime::now(), transactions);
+        let mut txn = subject.more_money_received(time_of_change, transactions.as_slice());
 
-        let status = subject.account_status(&debtor);
-        assert!(status.is_none());
+        txn.commit().unwrap();
+        drop(txn);
+        let actual_record_1 = subject.account_status(&first_tracked_wallet).unwrap();
+        assert_eq!(actual_record_1.wallet, first_tracked_wallet);
+        assert_eq!(
+            actual_record_1.balance_wei,
+            first_initial_balance as i128 - 1111
+        );
+        assert_eq!(
+            to_time_t(actual_record_1.last_received_timestamp),
+            to_time_t(time_of_change)
+        );
+        let actual_record_2 = subject.account_status(&unknown_wallet);
+        assert!(actual_record_2.is_none());
+        let actual_record_3 = subject.account_status(&second_tracked_wallet).unwrap();
+        assert_eq!(actual_record_3.wallet, second_tracked_wallet);
+        assert_eq!(
+            actual_record_3.balance_wei,
+            second_initial_balance as i128 - 9999
+        );
+        assert_eq!(
+            to_time_t(actual_record_3.last_received_timestamp),
+            to_time_t(time_of_change)
+        );
+        let log_handler = TestLogHandler::new();
+        log_handler.exists_log_containing(&format!(
+            "INFO: {test_name}: Detected an inward transfer of 2222 wei in the block 4446. \
```

Comment:
I don't know whether it's a good suggestion, but instead of logging shouldn't we create a new table to store txns from unknown wallets.

Reply:
Nobody bore this out, showing interest, so I chose not to do it.